pennylane/qnode.py

```diff
@@ -29,6 +29,24 @@
 from pennylane.tape import QuantumTape, make_qscript
 
 
+def _convert_to_interface(res, interface):
+    """
+    Recursively convert res to the given interface.
+    """
+    interface = INTERFACE_MAP[interface]
+
+    if interface in ["Numpy", "autograd"]:
+        return res
+
+    if isinstance(res, (list, tuple)):
+        return type(res)(_convert_to_interface(r, interface) for r in res)
+
+    if isinstance(res, dict):
+        return {k: _convert_to_interface(v, interface) for k, v in res.items()}
+
+    return qml.math.asarray(res, like=interface if interface != "tf" else "tensorflow")

```

Comment:
This would araises backprop information on tensors with PyTorch, so it can potentially cause some issue in a bigger computation. Could you instead call it only if the QNode has no parameters?